src/AccountIcon.jsx

```diff
@@ -1,45 +1,50 @@
 const blockies = require('blockies');
 const React = require('react');
-const {Image} = require('semantic-ui-react');
-const {ReactiveComponent} = require('oo7-react');
+const PropTypes = require('prop-types');
+
+const { Image } = require('semantic-ui-react');
+const { ReactiveComponent } = require('oo7-react');
 
 function createIdentityImage (address, scale = 8) {
 	return blockies({
-    	seed: (address || '').toLowerCase(),
-    	size: 8,
-    	scale
-    }).toDataURL();
+		seed: (address || '').toLowerCase(),
+		size: 8,
+		scale
+	}).toDataURL();
 }
 
 class AccountIcon extends ReactiveComponent {
-	constructor() { super(['address', 'className', 'style']); }
+	constructor () { super(['address', 'className', 'style']); }
 
-	render() {
-		if (typeof(this.state.address) == "string") {
-			return (<Image
-				inline
-				avatar={this.props.avatar}
-                src={createIdentityImage(this.state.address)}
-				style={Object.assign({borderRadius: '50%'}, this.state.style)}
-                className={typeof(this.state.className) === 'string' ? this.state.className : ''}
-                id={this.props.id}
-                data-address-img
-            />);
+	render () {
+		if (typeof (this.state.address) === 'string') {
+			return (
+				<Image
+					inline
+					avatar={ this.props.avatar }
+					src={ createIdentityImage(this.state.address) }
+					style={ Object.assign({ borderRadius: '50%' }, this.state.style) }
+					className={ typeof (this.state.className) === 'string' ? this.state.className : '' }
```

Comment:
Another example of unneeded typeof. Here you could write `className={this.state.className}` directly, because in the propTypes we already enforce className to be a string.

Look for instances in the code like this, and try to remove unnecessary typeof.

Reply:
the type of the state is not being validated anywhere other than this, so I'd imagine we need to validate